Domain Connect Spec Draft.adoc

```diff
@@ -1388,6 +1388,17 @@ SRV: name, target, protocol, service, priority, weight, port, TTL
 |This OPTIONAL parameter identifies the group
 the record belongs to when applying changes.
 
+[[essential-record]]
+|*Essential*
+|enum
+|essential
+|(optional)
+This parameter indicates, whether record is obligatory or optional during template application and whether it can be removed without the whole template after application
+

```

Comment:
I would consider calling this something else.  This is really about impacting the conflict detection logic.

Three ideas:

One, make it a boolean.  Call it "TriggerConflicts". Default (True) means alterations in this record trigger conflicts.  False means skip.

Alternativey call it "SkipConflicts" as a bool. I awlays like a bool to have a default of false, not true.

If you want an enum, call it "TriggerConflicts".  Values could be "Always" or "Never".

If you are married to essentials, I'm OK.  Just trying to be clear.

Reply:
Hi @arnoldblinn , not really married however SkipConflicts = True or especially TriggerConflicts=Never would mean also no conflicts detection when applying such record as well.
In this sense essential=OnApply is for me a direct expression of what it is. May be even more explicit with "OnAppluOnly".

Reply:
I'm OK with your names.

Let's make the text:

This parameter indicates how the record should be treated during conflict detection (if the DNS Provider is not implementing Conflict Detection it is ignored).

Reply:
Change commited to 25e3264